test/cmocka/src/math/arithmetic/exponential_hifi.c

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <math.h>
+#include <cmocka.h>
+#include <string.h>
+
+#include <sof/math/exp_fcn_hifi.h>
+#include <sof/audio/format.h>
+#include <rtos/string.h>
+#include <sof/common.h>
+
+#define ULP_TOLERANCE 5.60032793
+#define ULP_SCALE 0.0000999999999749
+/* testvector in Q4.28, -5 to +5 */
+#if (EXPONENTIAL_HIFI3 == 1 || EXPONENTIAL_HIFI4 == 1 EXPONENTIAL_HIFI5 == 1)
+	static const int32_t iv[128] = {
+	    -1342177280, -1321040630, -1299903980, -1278767330, -1257630680, -1236494030,
+	    -1215357380, -1194220729, -1173084079, -1151947429, -1130810779, -1109674129,
+	    -1088537479, -1067400829, -1046264179, -1025127529, -1003990879, -982854229,
+	    -961717579,	 -940580929,  -919444278,  -898307628,	-877170978,  -856034328,
+	    -834897678,	 -813761028,  -792624378,  -771487728,	-750351078,  -729214428,
+	    -708077778,	 -686941128,  -665804477,  -644667827,	-623531177,  -602394527,
+	    -581257877,	 -560121227,  -538984577,  -517847927,	-496711277,  -475574627,
+	    -454437977,	 -433301327,  -412164677,  -391028026,	-369891376,  -348754726,
+	    -327618076,	 -306481426,  -285344776,  -264208126,	-243071476,  -221934826,
+	    -200798176,	 -179661526,  -158524876,  -137388226,	-116251575,  -95114925,
+	    -73978275,	 -52841625,   -31704975,   -10568325,	10568325,    31704975,
+	    52841625,	 73978275,    95114925,	   116251575,	137388226,   158524876,
+	    179661526,	 200798176,   221934826,   243071476,	264208126,   285344776,
+	    306481426,	 327618076,   348754726,   369891376,	391028026,   412164677,
+	    433301327,	 454437977,   475574627,   496711277,	517847927,   538984577,
+	    560121227,	 581257877,   602394527,   623531177,	644667827,   665804477,
+	    686941128,	 708077778,   729214428,   750351078,	771487728,   792624378,
+	    813761028,	 834897678,   856034328,   877170978,	898307628,   919444278,
+	    940580929,	 961717579,   982854229,   1003990879,	1025127529,  1046264179,
+	    1067400829,	 1088537479,  1109674129,  1130810779,	1151947429,  1173084079,
+	    1194220729,	 1215357380,  1236494030,  1257630680,	1278767330,  1299903980,
+	    1321040630,	 1342177280};
+#endif
+
+static void test_math_arithmetic_exponential_hifi4_fixed(void **state)
+{
+	(void)state;
+#if (EXPONENTIAL_HIFI3 == 1 || EXPONENTIAL_HIFI4 == 1 EXPONENTIAL_HIFI5 == 1)
```

Comment:
#if (EXPONENTIAL_HIFI3 == 1 || EXPONENTIAL_HIFI4 == 1 || EXPONENTIAL_HIFI5 == 1)?

Reply:
Thanks.  I 'II fix .

Reply:
There should be a top level CONFIG_MATH_EXP in Kconfig and then the architecture level HIFI kconfigs can be used.

Reply:
Done !! I've incorporated your suggestion. Should I change others, such as `CONFIG_SQRT_FIXED, CONFIG_CORDIC_FIXED,` and so on?